pulsar-broker/src/main/java/org/apache/pulsar/broker/service/PrecisPublishLimiter.java

```diff
@@ -63,6 +67,14 @@ public boolean isPublishRateExceeded() {
         return false;
     }
 
+    private void releaseThrottle() {
+        for (RateLimiter rateLimiter : rateLimiters.values()) {
+            if (rateLimiter.getAvailablePermits() <= 0) {
+                return;
+            }
+        }
+        this.rateLimitFunction.apply();
+    }
```

Comment:
Good catch! It seems that this logic is the "beef" of this PR and resolves the issue. 

Would it be possible to create a minimal PR with just these changes without any change to use a dynamic hashmap to store rate limiters? Just use the  `topicPublishRateLimiterOnMessage` and `topicPublishRateLimiterOnByte` fields directly.

Think of a good name for the method `releaseThrottle` isn't very descriptive. Please also add comments to explain the logic.

Reply:
This has been resolved in the latest commit.